runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/GrpcWindmillServer.java

```diff
@@ -1412,4 +1451,83 @@ public void close() throws IOException {
       stream.close();
     }
   }
+
+  /**
+   * A set of stopwatches used to track the amount of time spent throttled due to Resource Exhausted
+   * errors. Throttle time is cumulative for all three rpcs types but not for all streams. So if
+   * GetWork and CommitWork are both blocked for x, totalTime will be 2x. However, if 2 GetWork
+   * streams are both blocked for x totalTime will be x. All methods are thread safe.
+   */
```

Comment:
I think using separate ThrottleTimer, one for each of GetWork, GetData, CommitWork, would lead to simpler ThrottleTimer logic. Then GrpcWindmillServer::getAndResetThrottleTime() can dispatch to all 3 ThrottleTimers.

Reply:
Yep that is simpler.  Done.